### DISCUSSION TITLE
`next/image` compulsory `alt` attribute

### DISCUSSION THREAD

**Discussion by nico-bachner** [vercel/next.js#24707] [Ideas]
Currently, the `alt` attribute on the `next/image` is optional, but since the alt text is essential for accessibility, would it not be preferable to make it mandatory for `next/image`?

If throwing an error if the alt text is missing is too radical a change to start with, a first step might be to implement the requirement in the type definitions, so TypeScript users get a type error if they leave it out, and then if that works well, as a next step to generate a warning if it's missing for regular JavaScript users as well.

I'd love to hear what others opinions on the matter are.

**Comment by gomflo** [comment]
Im not sure about making mandatory the `alt` attr as is out of scope of `next/image` because it's a W3 recommendation.

**Reply by nico-bachner** [reply]
One of the reason we use frameworks such as Next.js is to help us make better decisions when building for the web. My thinking was that Next could go above and beyond the w3c recommendation and force devs to adhere to a higher standard. This is already the philosophy behind [Svelte](https://github.com/sveltejs/svelte) — their compiler starts shouting at you if you don't adhere to the accessibility recommendations.

**Comment by bobarros** [comment]
Warning about best practices seems like a good idea to me.
Good practices as mandatory most of the time seem like a bad idea.

If you have to build a website using an existing image database with no alt, the website will end up with fake alt descriptions anyway. Wich is worse than when they don't exist, actually.

Bear in mind that people writing those descriptions not understanding how they are used. Everyone thinks that because they exist is good enough and the truth is that most are just useless.

Look how a fair alt looks like (from a big journal) and try to compare with most that you see. Now, ask people that are writing them if they know that this is the right way to do it.

_______________

_alt="Edward Stely walks through thousands of paper gun records at the Bureau of Alcohol, Tobacco, Firearms and Explosives in Martinsburg, W.V."_

_______________

**Reply by nico-bachner** [reply]
You raise a very good point. 

I hadn't thought of the potential edge cases, and you're right that in such a situation it's better to omit the alt entirely.

I guess, as you say, giving the developer a warning would be a better solution than making the `alt` tag mandatory.

**Comment by hendridg** [comment]
Hi guys, 

I know this is no referral about making mandatory alt text, but I think is necessary to place the alt text in the middle of the box, I have many problems with that, and my SEO manager pushed me to fixed, thx!
